packages/TelegramClient-UI.package/TCUEmojiHelper.class/class/promptDownload.st

```diff
@@ -0,0 +1,5 @@
+downloading
+promptDownload
+
+	UIManager default inform: 'Emojis are not present on your computer. Click Ok to download now.'.
+	[self assureEmojis] forkAt: Processor userBackgroundPriority.
```

Comment:
On Mac, I'm experiencing sever performance issues. I opened the Telegram Client, clicked "ok" to download the emojis and waited for a while. When I clicked on one of the chats, the image froze until I aborted one or two minutes later. If that is expected because downloading the emojis just takes so long: please create a loading bar during the download. If that is not expected, we have another critical bug 😛 

Apart from that, startup is also really slow. Zip unpacking is happening at about 1 file per second. Is that also related to downloading the emojis? I could imagine the userBackgroundPriority being the issue on mac, given that background priority always leads to performance issues here.

Reply:
i am not the one who put the `userBackgroundPriority` there, the fix would be to replace `forkAt:` with just a plain `fork`, right?

Reply:
Yes, replacing forkAt: with fork should work. With regards to the progress bar: a very rudimental one could look like this:

```
ensureEmojisPresent

	self areEmojisPresent ifFalse: [
		'Downloading Emojis'
			displayProgressAt: Display center
			from: 0
			to: 2
			during: [: bar |
					self downloadArchive.
					bar value: 1.
					self removeOptionalCharacters.
					bar value: 2.
				]
	].
```